src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/ExtendedScanResult.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.Generic;
+using Microsoft.ComponentDetection.Contracts.BcdeModels;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.Sbom.Adapters.ComponentDetection;
+
+[JsonObject(MemberSerialization.OptOut, NamingStrategyType = typeof(CamelCaseNamingStrategy))]
+public class ExtendedScanResult : ScanResult
+{
+    new public IEnumerable<ExtendedScannedComponent> ComponentsFound { get; set; }
+}
```

Comment:
nit:
```suggestion
public sealed class ScanResultWithLicense : ScanResult
{
    public new IEnumerable<ScannedComponentWithLicense> ComponentsFound { get; init; }
}
```

I'd prefer `WithLicense` instead of `Extended` if possible. It's clearer what it's doing.